src/Axiom/Extensionality/Propositional.agda

```diff
@@ -23,6 +23,14 @@ Extensionality a b =
   {A : Set a} {B : A → Set b} {f g : (x : A) → B x} →
   (∀ x → f x ≡ g x) → f ≡ g
 
+-- A variant for implicit function spaces.
+
+ExtensionalityImplicit : (a b : Level) → Set _
+ExtensionalityImplicit a b =
+  {A : Set a} {B : A → Set b} {f g : {x : A} → B x} →
+  (∀ {x} → f {x} ≡ g {x}) → (λ {x} → f {x}) ≡ (λ {x} → g {x})
```

Comment:
Do we want this `∀ {x} → f {x} ≡ g {x}` to be `∀ x → f {x} ≡ g {x}` instead?

Reply:
My reasoning is that if `f {x} ≡ g {x}` holds definitionally then
we don't need to use functional extensionality. If we do it's because
we need to inspect `x` to make the two sides equal!

Reply:
The use case that prompted this pull request was something like a less contrived version of the following:
```agda
f : (x : ℕ) -> {y : ℕ} -> ℕ
f x {y} = x + y

g : (x : ℕ) -> {y : ℕ} -> ℕ
g x {y} = y + x

f=g : f ≡ g
f=g = ext λ x → iext (+-comm x _)
```
My thought was that often we do not need to inspect `y`, we just need to go under the implicit binder in order to continue inspecting something else. Hopefully `y` should be inferable in what follows, because if we need to inspect `y` for the definition to reduce, why was it implicit to start with?